sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1188,4 +1188,42 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
         classOf[BroadcastNestedLoopJoinExec]))
     }
   }
+
+  test("SPARK-32399: Full outer shuffled hash join") {
+    val inputDFs = Seq(
+      // Test unique join key
+      (spark.range(10).selectExpr("id as k1"),
+        spark.range(30).selectExpr("id as k2")),
```

Comment:
Could you add test cases for multiple join keys, e.g., (short, short, int) keys?

Reply:
@maropu - sure. Added a test case for multiple join keys.